profiling/src/lib.rs

```diff
@@ -240,6 +241,36 @@ extern "C" fn minit(_type: c_int, module_number: c_int) -> ZendResult {
         trace!("MINIT({_type}, {module_number})");
     }
 
+    #[cfg(feature = "tracing-subscriber")]
+    {
+        use std::fs::File;
+        use std::os::fd::FromRawFd;
+        use std::sync::Mutex;
+
+        let fd = loop {
+            // SAFETY:
+            let result = unsafe { libc::dup(libc::STDERR_FILENO) };
+            if result != -1 {
+                break result;
+            } else {
+                let error = std::io::Error::last_os_error();
+                if error.kind() != std::io::ErrorKind::Interrupted {
+                    error!("failed duplicating stderr to create tracing subscriber: {error}");
+                    return ZendResult::Failure;
+                }
+            }
```

Comment:
note: Could in theory run infinite, but as this is only meant to be used for dev and benchmarking, I do not care. Just in case we plan to release this to customer builds, we should define an exit condition of the loop.

Reply:
It should only loop forever if the syscalls keep returning `Interrupted`, right?

Reply:
Yeah, which is highly unlikely, that's why I approved anyway.